### PR TITLE
🌱 Skip upsynced resources in resource scheduling

### PR DESCRIPTION
**Pull request by davidfestal**
## Summary

Resources upsynced from a downstream cluster, and thus having the `Upsync` label should not be taken in account by resource scheduling.

## Related issue(s)

No issue dedicated. Spotted while testing PR https://github.com/kcp-dev/kcp/pull/2338

### CODE REVIEW

File: pkg/reconciler/workload/resource/resource_reconcile.go [hunk 1]
```diff
@@ -47,6 +47,11 @@ func (c *Controller) reconcileResource(ctx context.Context, lclusterName logical
 	logger := logging.WithObject(logging.WithReconciler(klog.Background(), ControllerName), obj).WithValues("groupVersionResource", gvr.String(), "logicalCluster", lclusterName.String())
 	logger.V(4).Info("reconciling resource")
 
+	if isUpSynced(obj.GetLabels()) {
+		logger.Info("resource is in Upsync mode; ignoring")
```
Comment (sttts) [inline pkg/reconciler/workload/resource/resource_reconcile.go:51]: verbosity at least 3, probably more like 4.

File: pkg/reconciler/workload/resource/resource_controller.go [hunk 2]
```diff
@@ -486,6 +486,17 @@ func (c *Controller) enqueueSyncTargetKey(syncTargetKey string) {
 	}
 }
 
+// isUpSynced returns true if the labels of the resource contain at least
+// one `ResourceState` label with the `Upsync` value.
+func isUpSynced(labels map[string]string) bool {
```
Comment (sttts) [inline pkg/reconciler/workload/resource/resource_controller.go:491]: as a special helper only used in one stop, please move it into the other file.

Comment (davidfestal) [reply pkg/reconciler/workload/resource/resource_controller.go]: fixed

File: pkg/reconciler/workload/resource/resource_reconcile.go (hunk 1 above)
Comment (davidfestal) [reply pkg/reconciler/workload/resource/resource_reconcile.go]: fixed

**Pull request merged**